stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/OfflineSyncFirebaseMessagingService.kt

```diff
@@ -58,41 +51,37 @@ internal class OfflineSyncFirebaseMessagingService : FirebaseMessagingService()
 
         GlobalScope.launch(DispatcherProvider.IO) {
             val cid: String = firebaseMessageParser.parse(message).let { "${it.channelType}:${it.channelId}" }
-            if (ChatDomain.isInitialized) {
-                performSync(ChatDomain.instance(), cid)
-
-                if (ChatClient.isInitialized) {
-                    ChatClient.instance()
-                        .onMessageReceived(message)
-                }
-            }
-
-            if (ChatDomain.isInitialized.not() || ChatClient.isInitialized.not()) {
+            if (ChatDomain.isInitialized && ChatClient.isInitialized) {
+                performSync(ChatDomain.instance(), cid, ChatClient.instance(), message)
+            } else {
                 val syncConfig = syncModule.encryptedBackgroundSyncConfigStore.get()
-
                 syncConfig?.let {
                     val config = it
                     val user = User(id = config.userId)
-                    val client = initClient(
-                        this@OfflineSyncFirebaseMessagingService,
-                        user,
-                        config.userToken,
-                        config.apiKey
-                    )
+                    val client = initClient(this@OfflineSyncFirebaseMessagingService, config.apiKey)
                     val domain = initDomain(user, client)
-                    performSync(domain, cid)
-                    client.onMessageReceived(message)
+                    logger.logD("Starting the sync, config: $syncConfig")
```

Comment:
I think we need to use the token from syncConfig after process is recreated. Currently, when message arrives, I see fallback notification.